src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,653 @@
+
+    async def test_create_category_invalid_entry_name_none_for_default(self):
+        """
+        Test that TypeError is returned when entry_name is None
+        :Assert:
+            Assert TypeError when default is None
+        """
+        with pytest.raises(TypeError) as error_exec:
+            await create_category(category_name='boolean', category_description='boolean type',
+                                  category_value={'info': {
+                                      'description': 'boolean type with default False ',
+                                      'type': 'boolean',
+                                      'default': None
+                                  }})
+        assert ("TypeError: entry_val must be a string for item_name info " +
+                "and entry_name default") in str(error_exec)
+
+    async def test_set_category_item_value_entry(self):
+        """
+        Test updating of configuration.value for a specific key
+        :assert:
+            1. `default` and `value` in configuration.value are the same
+            2. `value` in configuration.value gets updated, while `default` does not
+        """
+        select_value_stmt = sa.select([_configuration_tbl.c.value]).select_from(
+            _configuration_tbl).where(_configuration_tbl.c.key == 'boolean')
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}})
+        try:
+            async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                async with engine.acquire() as conn:
+                    async for result in conn.execute(select_value_stmt):
+                        assert result[0]['info']['value'] == 'False'
+                        assert result[0]['info']['default'] == 'False'
+        except Exception:
+            raise
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        try:
+            async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                async with engine.acquire() as conn:
+                    async for result in conn.execute(select_value_stmt):
+                        assert result[0]['info']['value'] == 'True'
+                        assert result[0]['info']['default'] == 'False'
+        except Exception:
+            raise
+
+    async def test_set_category_item_value_error(self):
+        """
+        Test updating of configuration.value when category_name does not exist
+        :assert:
+            Nothing happens / returned
+        :expect:
+            FOGL-522: When updating category item value, when the category_name
```

Comment:
add xfail reason=FOGL-522 ?

Reply:
@sstefano69 Also, is FOGL-522 the correct jira?

Reply:
Since the behavior would change when FOGL-522 is fixed, I don't see the point of adding an `xfail` comment since the test (as it stands) currently passes. Once the test will fail, then putting `xfail`, and a signing someone to fix it  makes sense.

Reply:
1) FOGL-522 is not the correct ticket id.
2) instead assert as per expectation, let it fail xFail reason = bug id ..... and once fixed this test (removal of xfail) pass will ensure bug fixed. TDD